core/graphelier-service/models/orderbooks.go

```diff
@@ -267,3 +267,31 @@ func (orderbook *Orderbook) ApplyMessagesToDeltabook(messages []*Message, numMes
 
 	return
 }
+
+// TopBookPerXNano : Returns a list of the top book at every x nanosecond after applying each individual message in a list
+func (orderbook *Orderbook) TopBookPerXNano(messages []*Message, x uint64) (topbook []*Point) {
+	topbookLengthIndex := 0
+	initialTimestamp := messages[0].Timestamp
+	for _, message := range messages {
+		singleMsgSlice := []*Message{message}
+		orderbook.ApplyMessagesToOrderbook(singleMsgSlice)
+		if (message.Timestamp-initialTimestamp)%x == 0 {
```

Comment:
This will skip over a few points.
Ex:
message nanos: [1 2 5 6]
You'll only record the point at 6 when there should be a point at 3 and 6

Reply:
I see... I'm not sure i understand your solution mentioned previously, can you explain again please?